### ISSUE TITLE
updating the expired doc removes the doc completely on SGW and CBS

### ISSUE THREAD

**Issue by sridevi-15** [couchbase/sync_gateway#3796]
### Sync Gateway version
happening on SGW  2.0.0, 2.1.0, 2.1.1

### Operating system
Centos

### Config file
```
{
    "interface":":4984",
    "adminInterface": "0.0.0.0:4985",
    "maxIncomingConnections": 0,
    "maxCouchbaseConnections": 16,
    "maxFileDescriptors": 90000,
    "slowServerCallWarningThreshold": 500,
    "compressResponses": false,


    "logging": {"debug": {"enabled": true} },
    "databases":{
        "db":{
            "import_docs": "continuous",
            "enable_shared_bucket_access": true,


            "num_index_replicas": 0,
            "server":"http://192.168.33.10,192.168.33.11,192.168.33.12:8091",
            "bucket":"data-bucket",
            "username":"data-bucket",
            "password": "password"
        }
    }
}

```

### Log output
https://gist.github.com/sridevi-15/785eef12d05cfe3f2b63c5a578823807
### Steps to reproduce
1.  Set up 3 nodes in a cluster in CBS
2.  Set up sync gateway and start the sync gateway
3.  create user in sync-gateway
4. Create doc with expiry like 120 seconds and add doc using SGW API.
5. update the doc by maintaining same expiry value 
6. Get the doc within the expiry value i.e within 120 seconds

### Expected behavior
Verify doc can be retrieved successully 
### Actual behavior
doc gets deleted right away though it has some time to expire

Note : it works successfully if I set up one node in cluster in CBS

**Comment by adamcfraser** [comment]
The attached logs are showing a 500 error during the update step.  Is that expected?

**Comment by sridevi-15** [comment]
@adamcfraser , can you check these logs? These are actual logs 
https://gist.github.com/sridevi-15/f61468a545444ca604d59e6cc361b4ee

**Comment by sridevi-15** [comment]
Also these are traces of API calls with responses for Step 5 and step 6:

1. GET http://192.168.33.13:4984/db/exp_3?conflicts=true&revs=true&show_exp=true 200
2. doc value is ..  {u'_exp': u'2018-10-10T07:19:14Z', u'_rev': u'1-946da3d3ef71ee40c0f0121587cfda56', u'channels': [u'NBC', u'ABC'], u'updates': 0, u'_revisions': {u'start': 1, u'ids': [u'946da3d3ef71ee40c0f0121587cfda56']}, u'_id': u'exp_3'}
3. Updating http://192.168.33.13:4984/db/exp_3: 1 times
4.  json dumps doc  for updating the doc {"_exp": "2018-10-10T07:19:14Z", "_rev": "1-946da3d3ef71ee40c0f0121587cfda56", "random": "b898b538-8597-44fa-af05-06a64e07111e", "channels": ["NBC", "ABC"], "updates": 1, "_revisions": {"start": 1, "ids": ["946da3d3ef71ee40c0f0121587cfda56"]}, "_id": "exp_3"}
5. GET http://192.168.33.13:4984/db/exp_3?conflicts=true&revs=true&show_exp=true 403

**Comment by sridevi-15** [comment]
@adamcfraser , yes document exp_3 is created with expiry of 120 seconds. Then I tried to update the doc. 
 After I create the doc,  doc body looks like which is at step #2 in previous comment and then I update doc  . I pasted json body which is parsing while updating doc at step #4 in my previous comment.

Yes, that is json payload to update the doc . Basically when I do get on the doc, it already has _exp value as I created in the doc with expiry value. I updated the doc by adding/updating key value pairs like 'updates:1', 'random:b898b538-8597-44fa-af05-06a64e07111e' , all others remain same.
